### PR TITLE
Unittest/usecases

### PR DESCRIPTION
**Pull request by ishanvaghani**

### CODE REVIEW

**Comment by ishanvaghani** [comment]
@hieuwu can you please check and merge?

File: app/src/main/java/com/hieuwu/groceriesstore/domain/usecases/impl/UpdateCartItemUseCaseImpl.kt [hunk 1]
```diff
@@ -19,16 +19,20 @@ class UpdateCartItemUseCaseImpl @Inject constructor(
 ) : UpdateCartItemUseCase {
     override suspend fun updateLineItem(lineItemModel: LineItemModel) {
         withContext(ioDispatcher) {
-            productRepository.updateLineItemQuantityById(
-                lineItemModel.quantity!!,
-                lineItemModel.id!!
-            )
+            if (lineItemModel.quantity != null && lineItemModel.id != null) {
```
Comment (hieuwu) [inline app/src/main/java/com/hieuwu/groceriesstore/domain/usecases/impl/UpdateCartItemUseCaseImpl.kt:22]: Good one 👍

**Review by hieuwu — COMMENTED** [review]
Overall looks good. Could you please share a screenshot of the test result?

**Comment by ishan-vaghani** [comment]
Screenshot 2023-12-27 at 10 37 56 AM

**Comment by hieuwu** [comment]
> Screenshot 2023-12-27 at 10 37 56 AM

Great work. Thank you for your contribution.

**Pull request merged**